Operator docs covering ingress TLS options (#2145)
kroxylicious/kroxylicious: docs/modules/configuring/con-virtualkafkacluster-mtls.adoc

// file included in the following:
//
// kroxylicious-operator/assemblies/assembly-operator-secure-client-proxy-connection.adoc

[id='con-kafka-client-mtls-{context}']
= Mutual TLS configuration for client-to-proxy connections

You can configure a virtual cluster ingress to request or require that the Kafka client authenticates to the proxy using TLS.

> This is known as mutual TLS (mTLS), because both the client and the proxy authenticate each other using TLS.

Reviewer: ```suggestion
This configuration is known as mutual TLS (mTLS), because both the client and the proxy authenticate each other using TLS.
```

> .Example `VirtualKafkaCluster` configured so that the proxy will *require* Kafka clients to present a trusted TLS client certificate.

Reviewer: ```suggestion
.Example `VirtualKafkaCluster` configuration requiring clients to present a trusted certificate
```

Reviewer: Not sure we need VirtualKafkaCluster in the title, so we could make it shorter by removing, but we should be consistent with approach in doc

[source,yaml]
----
kind: VirtualKafkaCluster
metadata:
  # ...
spec:
  # ...
  ingresses:
    - ingressRef:
        name: cluster-ip
      tls:
        certificateRef:
          # ...
        trustAnchorRef: # <1>
          kind: ConfigMap # <2>
          name: trusted-cas # <3>
          key: trusted-cas.pem # <4>
        tlsClientAuthentication: REQUIRED <5>
----

> <1> The `trustAnchorRef` property references a separate Kubernetes resource which contains the CA certificates to be trusted.

Reviewer: ```suggestion
<1> References a separate Kubernetes resource containing the trusted CA certificates.
```

<2> The `kind` is optional and defaults to `ConfigMap`.

> <3> The `name` of the resource of the given `kind`. This resource must exist in the same namespace as the `VirtualKafkaCluster`

Reviewer: ```suggestion
<3> Name of the resource of the given `kind`, which must exist in the same namespace as the `VirtualKafkaCluster`.
```

> <4> The `key` identifies the entry in the given resource. The corresponding value must be a set of CA certificates. Supported formats for the bundle are: PEM, PKCS#12 and JKS.

Reviewer: ```suggestion
<4> Key identifying the entry in the given resource. The corresponding value must be a set of CA certificates. Supported formats for the bundle are: `PEM`, `PKCS#12`, and `JKS`.
```
